### PR TITLE
List entry inside expanded row of sources table

### LINKED ISSUE #1759: Tagging Page: Entries Listing
### New Addition
- Show list of entries on expanded row
- Paginate entries
- The list should be expanded on button click on count of entries as in the screenshot and not the whole row

### Attachments
- [Figma Link](https://www.figma.com/proto/HaQd4COVlpSMoqSfgL5ZDx/DEEP---UI_Library-%2B-Screens?page-id=522%3A417&node-id=6942%3A24982&viewport=13892%2C7980%2C0.31971144676208496&scaling=scale-down-width&starting-point-node-id=7917%3A4&hide-ui=1)

### PR DESCRIPTION
**Pull request by samshara**
- fixes #1759,  #1760, #1983, #1984,  #1755,  #1758

## Changes

* Pull entry list for the lead
* Paginate entries
* Add verification of entry feature
* Add control entry feature
* Add edit tags, show comments button

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: app/views/Project/Tagging/Sources/SourcesTable/EntryList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    const entryDataRendererParams = useCallback((_: string, data: Entry) => ({
+        entry: data,
+    }), []);
+
+    return (
+        <Container
+            className={_cs(className, styles.entryListContainer)}
+            contentClassName={styles.content}
+            footerActions={((entryListResponse?.count ?? 0) > maxItemsPerPage) && (
```
Comment (tnagorra) [inline app/views/Project/Tagging/Sources/SourcesTable/EntryList/index.tsx]: Any reason we are doing this? Pager wil automatically hide itself

**Review by tnagorra — DISMISSED** [review]

File: app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            headerClassName={styles.header}
+            contentClassName={styles.content}
+            headerActions={(
+                <EntryVerification
+                    entryId={entry.id}
+                    projectId={entry.project}
+                    onSuccess={handleChange}
+                    value={entry.verified}
+                    verifiedByCount={entry.verifiedByCount}
+                />
```
Comment (tnagorra) [inline app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/index.tsx]: Lets disable this if we have not implemented the behavior

File: app/Base/hooks/useProjectMemberListQuery.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { useRequest } from '#base/utils/restRequest';
```
Comment (tnagorra) [inline app/Base/hooks/useProjectMemberListQuery.ts]: I prefer not to create custom hooks for queries

File: app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryVerification/EntryUnverifyCommentModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        text: [requiredStringCondition],
+        mentionedUsers: [requiredListCondition],
+    }),
+};
+
+const defaultFormValue: FormType = {};
+
+export interface Props {
+    onModalClose: () => void;
+    onValidationSuccess: (value: FormType) => void;
```
Comment (tnagorra) [inline app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryVerification/EntryUnverifyCommentModal/index.tsx]: The modal handle form success.

**Comment by tnagorra** [comment]
Please fix these warnings
image

**Comment by tnagorra** [comment]
This doesn't look correct.
1. Only one entry is listed. But the total page size is 22610 (it's not possible to have that many entry in a lead)
2. The entry should be shown in readOnly mode
3. We should only expand the table if we click on the "Entries" button

image

File: app/types/newEntry.ts [hunk 5]
```diff
@@ -171,4 +171,31 @@ export interface Entry extends Entity {
...
+    textHistory: string[];
+    lead: number;
+    createdByDetails: {
+        id: number;
+        name: string;
+        email: string;
+        organization: string;
+        displayPictureUrl: string;
+    };
+    mentionedusersDetails: {
```
Comment (AdityaKhatri) [inline app/types/newEntry.ts]:
```suggestion
    mentionedUsersDetails: {
```

File: app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryControl/index.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        url: `server://v2/projects/${projectId}/project-memberships/`,
+        method: 'GET',
+        failureHeader: 'Project Membership',
+    });
+
+    const isQualityController = useMemo(() => (
+        projectMembersResponse?.results.find((v) => v.member.toString() === user?.id)
+            ?.badges.some((v) => (v === QUALITY_CONTROLLER))
+    ),
+    [projectMembersResponse?.results, user?.id]);
```
Comment (AdityaKhatri) [inline app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryControl/index.tsx]: @samshara , this should come from the ProjectContext.
@thenav56 , could you add `CAN_QUALITY_CONTROL` or similar permission in allowedPermissions?

File: app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryControl/index.tsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+    const controlStatusLabel = useMemo(() => {
+        if (isQualityController) {
+            return value
+                ? 'Uncontrol'
+                : 'Control';
+        }
+        return value
+            ? 'Controlled'
+            : 'Uncontrolled';
```
Comment (AdityaKhatri) [inline app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryControl/index.tsx]:
```suggestion
        return value
            ? 'Controlled'
            : 'Control';
```

File: app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryVerification/EntryUnverifyCommentModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+    MultiResponse,
+    Membership,
+} from '#types';
+
+import styles from './styles.css';
+
+const memberFieldQuery = {
+    fields: ['member', 'member_name'],
+};
+const UNVERIFY = 2;
```
Comment (AdityaKhatri) [inline app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryVerification/EntryUnverifyCommentModal/index.tsx]: Lets keep all the enums in a util file or something similar and fetch it from there.

File: app/views/Project/Tagging/Sources/SourcesTable/index.tsx [hunk 9]
```diff
@@ -390,15 +409,18 @@ function SourcesTable(props: Props) {
                 )}
             >
                 {pending && (<PendingMessage />)}
-                <SortContext.Provider value={sortState}>
```
Comment (AdityaKhatri) [inline app/views/Project/Tagging/Sources/SourcesTable/index.tsx:393]: We need SortContext here

File: app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/index.tsx [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        publishedOn: string;
+        createdByName: string;
+        authors: {
+            title: string;
+        }[];
+        source: {
+            title: string;
+        };
+    };
+}
```
Comment (AdityaKhatri) [inline app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/index.tsx]: This is not necessary

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryControl/index.tsx (hunk 6 above)
Comment (thenav56) [reply app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryControl/index.tsx]: Sure. @AdityaKhatri .

**Review by AdityaKhatri — DISMISSED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**